x-pack/plugins/ess_security/public/common/hooks/use_variation.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import type {
+  CloudExperimentsFeatureFlagNames,
+  CloudExperimentsPluginStart,
+} from '@kbn/cloud-experiments-plugin/common';
+/**
+ * Retrieves the variation of the feature flag if the cloudExperiments plugin is enabled.
+ * @param cloudExperiments {@link CloudExperimentsPluginStart}
+ * @param featureFlagName The name of the feature flag {@link CloudExperimentsFeatureFlagNames}
+ * @param defaultValue The default value in case it cannot retrieve the feature flag
+ * @param setter The setter from {@link useState} to update the value.
+ */
+export const useVariation = <Data>(
```

Comment:
This is re-created from x-pack/plugins/security_solution/public/common/components/utils.ts and still consumed by x-pack/plugins/security_solution/public/app/home/global_header/index.tsx

Reply:
We'll have to ask the proper team about this "variation" thing, the code looks temporary. That's okay for now